source/common/network/filter_manager_impl.cc

```diff
@@ -122,6 +189,11 @@ FilterStatus FilterManagerImpl::onWrite(ActiveWriteFilter* filter,
     return FilterStatus::StopIteration;
   }
 
+  // Only inject write is allowed when the connection is pending local close.
+  if (filter ? state_.remote_close_pending_ : pendingClose()) {
+    return FilterStatus::StopIteration;
+  }
```

Comment:
Sorry, I got the comment but didn't get the code. Could you explain this a little more? Thanks.

Reply:
Added more comments. Inject write is associated with a filter object, and we will stopIteration if a remote close is pending. For normal close situation, no more write is allowed.